components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -19,6 +19,7 @@ abstract class EngineSession {
         fun onLocationChange(url: String)
         fun onProgress(progress: Int)
         fun onLoadingStateChange(loading: Boolean)
+        fun onNavigationStateChange(canGoBack: Boolean?, canGoForward: Boolean?)
```

Comment:
What do you think about defining default values here:

```Kotlin
fun onNavigationStateChange(canGoBack: Boolean? = null, canGoForward: Boolean? = null)
```

With that you wouldn't need to pass null here...

```Kotlin
notifyObservers { onNavigationStateChange(null, canGoForward) }
```
... and instead you could just set the named parameter you actually have a value for:

```Kotlin
notifyObservers { onNavigationStateChange(canGoBack=...) }
```

Reply:
Good idea! I will update this.